failed to find dbghelp.dll Windows 7/Visual Studio 2010 no SDK

C:\drmemory\build>cmake -G"Visual Studio 10" -DCMAKE_BUILD_TYPE=Debug ..\drmemory
-- Dr. Memory version number: 1.11.17239
-- Targeting subsystem 5.00
Looking: /bin/x86/dbghelp.dll/tools/tracing/i386/dbghelp.dllC:\Program Files/Microsoft Visual Studio */Common7/IDE/Remote Debugger/x86/dbghelp.dllC:\Program Files (x86)/Windows Kits/*/Debuggers/x86/dbghelp.dllC:\Program Files/Microsoft Visual Studio */Common7/IDE/dbghelp.dllC:\Program Files/Debugging Tools for Windows/dbghelp.dll
CMake Error at CMakeLists.txt:597 (message):
  dbghelp.dll required and not found


-- Configuring incomplete, errors occurred!
See also "C:/drmemory/build/CMakeFiles/CMakeOutput.log".

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -550,12 +550,12 @@ if (WIN32)
     "${DDK_ROOT}/bin/${ARCH_SFX}/dbghelp.dll"
     "${DDK_ROOT}/tools/tracing/${DDK_SFX}/dbghelp.dll"
     "${PROGFILES}/Microsoft Visual Studio */Common7/IDE/Remote Debugger/${ARCH_SFX}/dbghelp.dll"
-    "${PROGFILES32}/Windows Kits/*/Debuggers/${ARCH_SFX}/dbghelp.dll")
+    "${PROGFILES32}/Windows Kits/*/Debuggers/${ARCH_SFX}/dbghelp.dll"
+    # For Visual Studio 2010+, x64 dbghelp.dll resides in Program
+    # Files (x86) rather than Program Files.
+    "${PROGFILES32}/Microsoft Visual Studio */Common7/IDE/Remote Debugger/${ARCH_SFX}/dbghelp.dll")
   if (X64)
     set(dbghelp_paths ${dbghelp_paths}
-      # For Visual Studio 2010+, x64 dbghelp.dll resides in Program
-      # Files (x86) rather than Program Files.
-      "${PROGFILES32}/Microsoft Visual Studio */Common7/IDE/Remote Debugger/${ARCH_SFX}/dbghelp.dll"
       # For older versions of windbg, x64 dbghelp.dll resides here.
       "${PROGFILES}/Debugging Tools for Windows (x64)/dbghelp.dll")
   else (X64)
@@ -599,12 +599,12 @@ if (WIN32)
   endif ()
   set(symsrv_paths
     "${PROGFILES}/Microsoft Visual Studio */Common7/IDE/Remote Debugger/${ARCH_SFX}/symsrv.dll"
-    "${PROGFILES32}/Windows Kits/*/Debuggers/${ARCH_SFX}/symsrv.dll")
+    "${PROGFILES32}/Windows Kits/*/Debuggers/${ARCH_SFX}/symsrv.dll"
+    # For Visual Studio 2010+, x64 symsrv.dll resides in Program
+    # Files (x86) rather than Program Files.
+    "${PROGFILES32}/Microsoft Visual Studio */Common7/IDE/Remote Debugger/${ARCH_SFX}/symsrv.dll")
   if (X64)
     set(symsrv_paths ${symsrv_paths}
-      # For Visual Studio 2010+, x64 symsrv.dll resides in Program
-      # Files (x86) rather than Program Files.
-      "${PROGFILES32}/Microsoft Visual Studio */Common7/IDE/Remote Debugger/${ARCH_SFX}/symsrv.dll"
       # For older versions of windbg, x64 symsrv.dll resides here.
       "${PROGFILES}/Debugging Tools for Windows (x64)/symsrv.dll")
   else (X64)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -551,8 +551,8 @@ if (WIN32)
     "${DDK_ROOT}/tools/tracing/${DDK_SFX}/dbghelp.dll"
     "${PROGFILES}/Microsoft Visual Studio */Common7/IDE/Remote Debugger/${ARCH_SFX}/dbghelp.dll"
     "${PROGFILES32}/Windows Kits/*/Debuggers/${ARCH_SFX}/dbghelp.dll"
-    # For Visual Studio 2010+, x64 dbghelp.dll resides in Program
-    # Files (x86) rather than Program Files.
+    # In case if SDK is not installed and we have Visual Studio, dbghelp.dll may be found in the
+    # Visual Studio's directory in subfolder Remote Debugger/x86 or /x64 folders (xref i#1956).
     "${PROGFILES32}/Microsoft Visual Studio */Common7/IDE/Remote Debugger/${ARCH_SFX}/dbghelp.dll")
   if (X64)
     set(dbghelp_paths ${dbghelp_paths}
@@ -600,8 +600,8 @@ if (WIN32)
   set(symsrv_paths
     "${PROGFILES}/Microsoft Visual Studio */Common7/IDE/Remote Debugger/${ARCH_SFX}/symsrv.dll"
     "${PROGFILES32}/Windows Kits/*/Debuggers/${ARCH_SFX}/symsrv.dll"
-    # For Visual Studio 2010+, x64 symsrv.dll resides in Program
-    # Files (x86) rather than Program Files.
+    # In case if SDK is not installed and we have Visual Studio, symsrv.dll may be found in the
+    # Visual Studio's directory in subfolder Remote Debugger/x86 or /x64 folders (xref i#1956).
     "${PROGFILES32}/Microsoft Visual Studio */Common7/IDE/Remote Debugger/${ARCH_SFX}/symsrv.dll")
   if (X64)
     set(symsrv_paths ${symsrv_paths}

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -549,10 +549,12 @@ if (WIN32)
   set(dbghelp_paths
     "${DDK_ROOT}/bin/${ARCH_SFX}/dbghelp.dll"
     "${DDK_ROOT}/tools/tracing/${DDK_SFX}/dbghelp.dll"
-    "${PROGFILES}/Microsoft Visual Studio */Common7/IDE/Remote Debugger/${ARCH_SFX}/dbghelp.dll"
     "${PROGFILES32}/Windows Kits/*/Debuggers/${ARCH_SFX}/dbghelp.dll"
     # In case if SDK is not installed and we have Visual Studio, dbghelp.dll may be found in the
-    # Visual Studio's directory in subfolder Remote Debugger/x86 or /x64 folders (xref i#1956).
+    # Visual Studio's directory in subfolders Remote Debugger/x86 or Remote Debugger/x64 (xref i#1956).
+    # We look in both paths (Program Files and Program Files (x86)) because some versions of
+    # Visual Studio put dbghelp.dll in progfiles (especially 2008) and some in progfiles32.
+    "${PROGFILES}/Microsoft Visual Studio */Common7/IDE/Remote Debugger/${ARCH_SFX}/dbghelp.dll"
     "${PROGFILES32}/Microsoft Visual Studio */Common7/IDE/Remote Debugger/${ARCH_SFX}/dbghelp.dll")
   if (X64)
     set(dbghelp_paths ${dbghelp_paths}
@@ -598,10 +600,10 @@ if (WIN32)
     message(STATUS "Using ${DBGHELP_DLL}")
   endif ()
   set(symsrv_paths
-    "${PROGFILES}/Microsoft Visual Studio */Common7/IDE/Remote Debugger/${ARCH_SFX}/symsrv.dll"
     "${PROGFILES32}/Windows Kits/*/Debuggers/${ARCH_SFX}/symsrv.dll"
     # In case if SDK is not installed and we have Visual Studio, symsrv.dll may be found in the
-    # Visual Studio's directory in subfolder Remote Debugger/x86 or /x64 folders (xref i#1956).
+    # same place as dbghelp.dll (see comment for dbghelp.dll) (xref i#1956).
+    "${PROGFILES}/Microsoft Visual Studio */Common7/IDE/Remote Debugger/${ARCH_SFX}/symsrv.dll"
     "${PROGFILES32}/Microsoft Visual Studio */Common7/IDE/Remote Debugger/${ARCH_SFX}/symsrv.dll")
   if (X64)
     set(symsrv_paths ${symsrv_paths}

